src/Discord.Net.Core/Entities/Interactions/MessageCommandCreationProperties.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Discord
+{
+    /// <summary>
+    ///     A class used to create Message commands.
+    /// </summary>
+    public class MessageCommandCreationProperties
```

Comment:
Remove unnecessary properties like the type as the type is inferred from the name of the class. Also the description is not used?

Reply:
Description property removed. Type is used in command creation, and command interaction.

Reply:
Yes but you have specified 2 different classes for application commands, user and message. Shouldnt the type be inferred inside of the api method and set their?

Reply:
you're saying set the type in DiscordRestApiClient for the create/modify for user/message/slash commands and just use the class names on receive? (WebSocket/.../SocketInteraction.cs)

So don't bother with the type in the SocketUser/Message/SlashCommand?